internal/testhelpers/e2e_server.go

```diff
@@ -62,7 +84,69 @@ func StartE2EServer(t *testing.T, configFile string) (publicUrl, adminUrl string
 			return fmt.Errorf("expected status code 200 but got: %d", res.StatusCode)
 		}
 		return nil
-	}), err)
+	}))
 
 	return
 }
+
+func CheckE2EServerOnHTTPS(t *testing.T, publicPort, adminPort int) (publicUrl, adminUrl string) {
+	publicUrl = fmt.Sprintf("https://127.0.0.1:%d", publicPort)
+	adminUrl = fmt.Sprintf("https://127.0.0.1:%d", adminPort)
+
+	require.NoError(t, retry.Do(func() error {
+		/* #nosec G402: TLS InsecureSkipVerify set true. */
+		tr := &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true}}
+		client := &http.Client{Transport: tr}
+		res, err := client.Get(publicUrl + "/health/alive")
+		if err != nil {
+			return err
+		}
+		defer res.Body.Close()
+		body := x.MustReadAll(res.Body)
+		if res.StatusCode != http.StatusOK {
+			t.Logf("%s", body)
+			return fmt.Errorf("expected status code 200 but got: %d", res.StatusCode)
+		}
+		return nil
+	}))
+
+	return
+}
+
+func GenerateTLSCertificateFilesForTests(t *testing.T, certPath, keyPath string) {
+	privateKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	assert.Nil(t, err)
+
+	cert, err := tlsx.CreateSelfSignedCertificate(privateKey)
+	assert.Nil(t, err)
```

Comment:
To ensure no panics etc:

```suggestion
	require.NoError(t, err)
```